tracing-opentelemetry-http/src/main/java/io/micronaut/tracing/opentelemetry/instrument/http/server/OpenTelemetryServerFilter.java

```diff
@@ -0,0 +1,126 @@
+import io.micronaut.http.MutableHttpResponse;
+import io.micronaut.http.annotation.Filter;
+import io.micronaut.http.filter.HttpServerFilter;
+import io.micronaut.http.filter.ServerFilterChain;
+import io.micronaut.tracing.opentelemetry.instrument.http.AbstractOpenTelemetryFilter;
+import io.micronaut.tracing.opentelemetry.instrument.util.OpenTelemetryExclusionsConfiguration;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import jakarta.inject.Named;
+import org.jetbrains.annotations.NotNull;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscription;
+import reactor.core.CoreSubscriber;
+
+import static io.micronaut.tracing.opentelemetry.instrument.http.server.OpenTelemetryServerFilter.SERVER_PATH;
+
+/**
+ * An HTTP server instrumentation filter that uses Open Telemetry.
+ *
+ * @author Nemanja Mikic
+ * @since 4.1.0
+ */
+@Filter(SERVER_PATH)
+@Requires(beans = Tracer.class)
+public class OpenTelemetryServerFilter extends AbstractOpenTelemetryFilter implements HttpServerFilter {
+
+    private static final String APPLIED = OpenTelemetryServerFilter.class.getName() + "-applied";
+    private static final String CONTINUE = OpenTelemetryServerFilter.class.getName() + "-continue";
+
+    private final Instrumenter<HttpRequest<?>, HttpResponse<?>> instrumenter;
+
+    /**
+     * @param exclusionsConfig The {@link OpenTelemetryExclusionsConfiguration}
+     * @param instrumenter The {@link OpenTelemetryHttpServerConfig}
+     */
+    public OpenTelemetryServerFilter(@Nullable OpenTelemetryExclusionsConfiguration exclusionsConfig,
+                                     @Named("micronautHttpServerTelemetryInstrumenter") Instrumenter<HttpRequest<?>, HttpResponse<?>> instrumenter) {
+        super(exclusionsConfig == null ? null : exclusionsConfig.exclusionTest());
+
+        this.instrumenter = instrumenter;
+    }
+
+    @Override
+    public Publisher<MutableHttpResponse<?>> doFilter(HttpRequest<?> request, ServerFilterChain chain) {
+        boolean applied = request.getAttribute(APPLIED, Boolean.class).orElse(false);
+        boolean continued = request.getAttribute(CONTINUE, Boolean.class).orElse(false);
+        if ((applied && !continued) || shouldExclude(request.getPath())) {
+            return chain.proceed(request);
+        }
+
+        request.setAttribute(APPLIED, true);
+
+        Publisher<MutableHttpResponse<?>> requestPublisher = chain.proceed(request);
+
+        return (Publishers.MicronautPublisher<MutableHttpResponse<?>>) actual -> {
+            Context parentContext = Context.current();
+            Context context = instrumenter.start(parentContext, request);
+
+            try (Scope ignored = context.makeCurrent()) {
+                requestPublisher.subscribe(new CoreSubscriber<MutableHttpResponse<?>>() {
```

Comment:
I think you still need to override `currentContext` if I remember. @dstepanov please provide guidance here